src/librustdoc/doctest.rs

```diff
@@ -349,6 +349,7 @@ fn run_test(
     compiler.env("UNSTABLE_RUSTDOC_TEST_PATH", path);
     compiler.env("UNSTABLE_RUSTDOC_TEST_LINE", format!("{}", line as isize - line_offset as isize));
     compiler.arg("-o").arg(&output_file);
+    compiler.arg("-W").arg("elided_lifetimes_in_paths");
```

Comment:
This enables the lint for _everyone_ using rustdoc, which is not what we want here. You should instead add it here as described in the issue, so it's only enabled in this repository. You also don't need the UI test then.
https://github.com/rust-lang/rust/blob/f94c926aecf226a12ffc5c7f09156390dde545ca/src/bootstrap/builder.rs#L1662-L1667

Reply:
> This enables the lint for everyone using rustdoc, which is not what we want here.

I wasn't sure about that. Thanks for clarification.

Reply:
Regarding my posted solution, I was wrong and misread the issue, adding it to the bootstrap builder doesn't help.